Fixed carbon dioxide going bank into tanks.

DeeJayhX opened the pull request:
# Pull Request Template

### Purpose

No one can breath carbon dioxide back into a 2000 kPa tank... it expels out into the atmosphere...

### Notes:

Does what it says on the tin.
### Please make sure you have followed the self checks below before submitting a PR:

- [x] - Code is sufficiently commented
- [x] - Code is indented with tabs and not spaces
- [x] - The PR does not include any unnecessary .meta, .prefab or <b>.unity (scene) changes</b>
- [x] - The PR does not bring up any new compile errors
- [x] - The PR has been tested in editor
- [x] - Any new files are named using PascalCase (to avoid issues on case sensitive file systems)
- [x] - Any new / changed components follow the [Component Development Checklist](https://github.com/unitystation/unitystation/wiki/Component-Development-Checklist)
- [x] - Any new objects / items follow the [Creating Items and Objects Guide](https://github.com/unitystation/unitystation/wiki/Creating-Items-and-Objects%3A-Now-With-Prefab-Variants) (especially concerning the use of prefab variants)
- [x] - The PR has been tested in multiplayer (with 2 clients and late joiners, if applicable)
- [x] - The PR has been tested with round restarts.
- [x] - The PR has been tested on moving / rotating / rotated-before-joining matrices (if applicable)

DeeJayhX committed:
Fixed carbon dioxide going bank into tanks.

No one can breath carbon dioxide back into a 2000 kPa tank... it expels out into the atmosphere...

chairbender commented on UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs:
```diff
@@ -119,7 +119,7 @@ private bool Breathe(IGasMixContainer node)
 		{
 			breathGasMix.RemoveGas(Gas.Oxygen, oxygenUsed);
-			breathGasMix.AddGas(Gas.CarbonDioxide, oxygenUsed);
+			node.GasMix.AddGas(Gas.CarbonDioxide, oxygenUsed);
```
Any time you change the gasmix on a node I am pretty sure you need to call MetaDataLayer.UpdateSystemsAt(position), otherwise atmos won't reliably detect the change. See how it is done in GasContainer.CheckRelease()

DeeJayhX replied:
Fixed.

chairbender reviewed (changes requested):
see comment

DeeJayhX committed:
Added MetaDataLayer Systems Update

-Added MetaDataLayer Systems Update
--To make sure that atmos will reliably detect the change.

fomalsd commented on UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs:
```diff
@@ -113,13 +113,17 @@ private bool Breathe(IGasMixContainer node)
 		{
 			breathGasMix.RemoveGas(Gas.Oxygen, oxygenUsed);
 			node.GasMix.AddGas(Gas.CarbonDioxide, oxygenUsed);
+			metaDataLayer.UpdateSystemsAt(position);
```
one-line and more reliable solution to what you're trying to achieve would be:
```suggestion
			MatrixManager.Get(playerScript.PlayerSync.ServerState.MatrixId).MetaDataLayer.UpdateSystemsAt(playerScript.PlayerSync.ServerLocalPosition);

```
so it'd be taking current matrix info from player's state.
also please don't use `transform.position` or `transform.localPosition` unless you're working on clientside lerping and such:
![img](https://user-images.githubusercontent.com/10403536/74655717-ba383480-5184-11ea-9cc1-d820173c6549.png)

DeeJayhX replied:
Fixed.

fomalsd reviewed (changes requested):
please don't rely on transform.positions for accurate position info, since it can be different from registered position on server

DeeJayhX committed:
Fixed tile location for MetaData update
